x/staking/keeper/val_state_change.go

```diff
@@ -24,7 +24,10 @@ func (k Keeper) BlockValidatorUpdates(ctx sdk.Context) []abci.ValidatorUpdate {
 	// unbonded after the Endblocker (go from Bonded -> Unbonding during
 	// ApplyAndReturnValidatorSetUpdates and then Unbonding -> Unbonded during
 	// UnbondAllMatureValidatorQueue).
-	validatorUpdates := k.ApplyAndReturnValidatorSetUpdates(ctx)
+	validatorUpdates, err := k.ApplyAndReturnValidatorSetUpdates(ctx)
+	if err != nil {
+		panic(err)
```

Comment:
I'm generally okay with the panics you added elsewhere, but this one inside the keeper seems risky. Though I don't have enough context to know what the risk really is.

At the very least, maybe a comment here explaining why a panic is okay here?